app/adapters/three_scale/api/instrumented_http_client.rb

```diff
@@ -57,8 +58,13 @@ def parse(response)
 
   protected
 
+  def zync_token
+    authentication = Rails.application.secrets.authentication || {}.freeze
+    authentication.fetch(:token) { return }
+  end
+
   def headers
-    super.merge('X-Request-Id' => SecureRandom.uuid)
+    super.merge('X-Request-Id' => SecureRandom.uuid, 'X-Zync-Token' => zync_token)
```

Comment:
There is a bug here. When `zync_token` returns nil, we are sending this header anyway and `build_request` does inside a `headers.each do |key, value| ... value.strip ... end` where that value is expected to be an String.
We can fix it by either returning `''` instead of nil or by not sending this header at all when it is empty, which is what makes more sense to me but both seem good to me.